tests/test_characteristic.py

```diff
@@ -202,3 +202,162 @@ def test_from_dict():
     assert char.display_name == 'Test Char'
     assert char.type_id == uuid
     assert char.properties == {'Format': 'int', 'Permissions': 'read'}
+
+class TestCharacteristic(unittest.TestCase):
```

Comment:
All tests in the `TestCharacteristic` class got changed in #99 as well. They are now outside, so you should be able to remove the class.